### DISCUSSION TITLE
notebook progress bar counts

### DISCUSSION THREAD

**Discussion by truekonrads** [tqdm/tqdm#1214] [Q&A] [answered]
- [X] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [X] visual output bug
- [X] I have visited the [source website], and in particular
  read the [known issues]
- [X] I have searched through the [issue tracker] for duplicates
- [ ] I have mentioned version numbers, operating system and
  environment, where applicable:
    4.59.0 3.8.8 (default, Apr 13 2021, 15:08:03) [MSC v.1916 64 bit (AMD64)] win32

I have a two progress bar setup. The outer progress bar ("t") works as expected, takes total, gets updated. But the inner progress bar does not: instead of displaying the configured total, it displays question mark. Instead of using "update" it updates based on some counter which appears to count every time time a yield is called inside the loop. The total is displayed correctly if no "update" call is made.

![image](https://user-images.githubusercontent.com/1421150/121982102-788fbc00-cdc1-11eb-9f70-7d6d2855d868.png)

[source website]: https://github.com/tqdm/tqdm/
[known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
[issue tracker]: https://github.com/tqdm/tqdm/issues?q=

**Comment by casperdcl** [comment, marked as answer]
replace `t2.update(posm)` with `t2.update(len(l)/(1024*1024))`

**Comment by era-net** [comment]
Hi

Shalltear ***@***.***> schrieb am Mi. 5. Jan. 2022 um 12:39:
